backend/src/Notifo.Domain.Integrations/Discord/DiscordIntegration.Messaging.cs

```diff
@@ -0,0 +1,101 @@
+﻿// =====================================================
+//  Notifo.io
+// ==========================================================================
+//  Copyright (c) Sebastian Stehle
+//  Author of the file: Artur Nowak
+//  All rights reserved. Licensed under the MIT license.
+// ==========================================================================
+
+using Discord;
+using Discord.Net;
+
+namespace Notifo.Domain.Integrations.Discord;
+public sealed partial class DiscordIntegration : IMessagingSender
+{
+    private const int Attempts = 5;
+    public const string DiscordChatId = nameof(DiscordChatId);
+
+    public void AddTargets(IDictionary<string, string> targets, UserInfo user)
+    {
+        var userId = GetUserId(user);
+
+        if (!string.IsNullOrWhiteSpace(userId))
+        {
+            targets[DiscordChatId] = userId;
+        }
+    }
+
+    public async Task<DeliveryResult> SendAsync(IntegrationContext context, MessagingMessage message,
+    CancellationToken ct)
+    {
+        if (!message.Targets.TryGetValue(DiscordChatId, out var chatId))
+        {
+            return DeliveryResult.Skipped();
+        }
+
+        return await SendMessageAsync(context, message, chatId, ct);
+    }
+
+    private async Task<DeliveryResult> SendMessageAsync(IntegrationContext context, MessagingMessage message, string chatId,
+       CancellationToken ct)
+    {
+        var botToken = BotToken.GetString(context.Properties);
+
+        for (var i = 1; i <= Attempts; i++)
+        {
+            try
+            {
+                var client = discordBotClientPool.GetDiscordClient(botToken, ct);
```

Comment:
Either the client should be disposed in the case of an error or we should put it outside of the loop, which makes probably more sense.